GsaGHTests/2_GooWrappers/GsaBool6GooTest.cs

```diff
@@ -1,59 +1,144 @@
-      Assert.Equal(expectedZ, output.Value.Z);
-      Assert.Equal(expectedXx, output.Value.Xx);
-      Assert.Equal(expectedYy, output.Value.Yy);
-      Assert.Equal(expectedZz, output.Value.Zz);
+      Assert.True(output.Value.Equals(value));
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void CastFromBoolShouldSetAllToTheSameBoolForRestrains(bool value) {
+      var param = new GsaRestraintParameter();
+      param.CreateAttributes();
+      ComponentTestHelper.SetInput(param, new GH_Boolean(value));
+      var output = (GsaBool6Goo)ComponentTestHelper.GetOutput(param);
+
+      Assert.True(output.Value.Equals(value));
+    }
+
+    [Fact]
+    public void NotTheSameTypeThereforeNotEqual() {
+      Assert.False(new GsaBool6().Equals(new Point3d()));
+    }
+
+    [Fact]
+    public void NullIsNotEqual() {
+      Assert.False(new GsaBool6().Equals(null));
+    }
+
+    [Fact]
+    public void ShouldCheckAllComponentWhenComparingWithBoolFalse() {
+      Assert.False(new GsaBool6(true, true, true, true, true, false).Equals(true));
     }
+
+    [Fact]
+    public void ShouldCheckAllComponentWhenComparingWithBoolTrue() {
+      Assert.True(new GsaBool6(true, true, true, true, true, true).Equals(true));
+    }
+
+    [Fact]
+    public void ShouldHaveTheSameHashCode() {
+      Assert.Equal(new GsaBool6(true, true, false, false, true, true).GetHashCode(),
+        new GsaBool6(true, true, false, false, true, true).GetHashCode());
+    }
+
+    [Theory]
+    [InlineData("free", true, true, true, true, true, true)]
+    [InlineData("pin", false, false, false, true, true, true)]
+    [InlineData("pinned", false, false, false, true, true, true)]
+    [InlineData("fix", false, false, false, false, false, false)]
+    [InlineData("fixed", false, false, false, false, false, false)]
+    [InlineData("release", true, true, true, true, false, false)]
+    [InlineData("released", true, true, true, true, false, false)]
+    [InlineData("hinge", true, true, true, true, false, false)]
+    [InlineData("hinged", true, true, true, true, false, false)]
+    [InlineData("charnier", true, true, true, true, false, false)]
+    [InlineData("rrrrrr", true, true, true, true, true, true)]
+    [InlineData("ffffff", false, false, false, false, false, false)]
+    [InlineData("ffffrr", false, false, false, false, true, true)]
+    public void CastFromStringTest(
+      string text, bool expectedX, bool expectedY, bool expectedZ, bool expectedXx, bool expectedYy, bool expectedZz) {
+      //release
+      var output = StringExtension.ParseRelease(text);
+      var releaseBool6 = new GsaBool6 {
+        X = expectedX,
+        Y = expectedY,
+        Z = expectedZ,
+        Xx = expectedXx,
+        Yy = expectedYy,
+        Zz = expectedZz
+      };
+      Assert.Equal(releaseBool6, output);
+
+      //restraint is the opposite of release
+      output = StringExtension.ParseRestrain(text);
+      Assert.Equal(!releaseBool6, output);
```

Comment:
as above

Reply:
@DominikaLos @psarras  Release/restraint is property of type bool6. It is single data class which hold 6 plain simple boolean data. I do not see reason why we should check x,y,z, xx, yy and zz in separate test